### ISSUE TITLE
`IfConditionRule` wrongly reports on usage of objects used with null coalescing operator (`??`)

### ISSUE THREAD

**Issue by ZebulanStanphill** [voku/phpstan-rules#18]
#### What is this feature about (expected vs actual behaviour)?

If you include a class in `classesNotInIfConditions`, that prevents you from using it with the null coalescing operator (`??`). I don't think this is expected behavior, nor do I think it's useful.

#### How can I reproduce it?

Add `\Apple\Banana\Thing` to your `classesNotInIfConditions` list in your PHPStan config and analyse this code:

```php
<?php declare(strict_types=1);
namespace Apple\Banana;

final class Thing {}

function doStuff(?Thing $maybeThing): Thing {
	$aThing = $maybeThing ?? new Thing();

	return $aThing;
}
```

#### Does it take minutes, hours or days to fix?

I figure it's simply a matter of ignoring a "comparison" if the operator is actually `??`. So hours, I guess.

**Comment by voku** [comment]
Can you maybe add a failing test as pull request, so that I can fix the problem more easily, thanks. 😊
